torch/csrc/autograd/variable.cpp

```diff
@@ -472,7 +472,7 @@ void handle_view_on_rebase(DifferentiableViewMeta* diff_view_meta, bool indirect
         TORCH_INTERNAL_ASSERT(false, "Invalid CreationMeta state");
       }
 
-      if (!indirect && !grad_fn) {
+      if (!indirect && !grad_fn && diff_view_meta->requires_grad()) {
```

Comment:
Does ` NOTE [ View + Inplace detection ] ` need updating now?

Reply:
I looked and it really doesn't explain very much. Perhaps there needs to be a comment here?

Reply:
I am willing to be convinced that this test is correct. However, I find this test surprising: how can it be that you (1) this is not a leaf, (2) have no grad-fn, and yet (3) somehow require grad? You say exactly what situation this can occur in the PR but I am having trouble translating this into an understanding of the new invariants on variable, after this change.

Reply:
Your point (1) is wrong I think: This is a leaf (equivalent to no grad-fn) and requires grad (in the cpp sense: either you require gradients or your _base does).
I'll add a link to the place where the leaf check is done to make it clearer.